__tests__/app.test.js

```diff
@@ -181,3 +183,75 @@ describe("GET /api/articles", () => {
       });
   });
 });
+
+describe("GET /api/articles/:articleid/comments", () => {
+  test("should return 200", () => {
+    return request(app).get("/api/articles/1/comments").expect(200);
+  });
+  test("should return an array of object that all the comments of the given article ", () => {
+    const articleId = 1
+    const count = commentData.reduce((counter, obj) => {
+      if (obj.article_id === articleId) counter += 1
+      return counter;
+    }, 0)
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .then(({body}) => {
+        expect(body.article.length).toEqual(count);
+      });
+  });
+  test("should return an array of object for the expected article id, each with the following keys: title, article_idtopic, author,created_at,votes,image_img_url,comment_count  ", () => {
+    const articleId = 1
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .then((response) => {
+        expect(response.body.article).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              comment_id: expect.any(Number),
+              votes: expect.any(Number),
+              created_at: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              article_id: articleId
+            }),
+          ])
+        );
+      });
+  });
+  test("should return an array of objects, sorted by created_at date by default where the most recent is on top ", () => {
+    const articleId = 1
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toBeSorted({ key: "created_at", descending:true });
+      });
+  });
+
+  test("should return  400 badRequest if provided id is invalid", () => {
+    const articleId = "invalid";
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          'Bad Request: ID provided is not valid'
+        );
+      });
+  });
+  test("should return  404 badRequest if provided id is non existent", () => {
+    const articleId = 9999;
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "Bad Request: ID provided has not been found."
+        );
+      });
+  });
+});
+
```

Comment:
Please add another test where it responds with status 200 and an empty array of comments.

Reply:
I have a few thoughts of this: the 404 test case  or returning a 404 is not the best practice for this endpoint. 
Technically this is not an error, it is not the endpoint's concern whether the id provided was valid or not. It will return a 200 with empty list, and 404 in most forums (Stackoverflow discussion) are not recommended. 
https://apihandyman.io/empty-lists-http-status-code-200-vs-204-vs-404/ 

It's function is to look up specifically for the article_id and return if there's an associated comment for it. In this case, it should be a 200. However I understand you can also look at it as invalid path given that the non-existent id is given as part of the path parameter. I have amended the code to reflect both cases: 
404 when article_id does not exist
200 when article_id exist, but no comments.